flexmeasures/utils/coding_utils.py

```diff
@@ -83,7 +83,10 @@ def flatten_unique(nested_list_of_objects: list) -> list:
         if isinstance(s, list):
             all_objects.extend(s)
         elif isinstance(s, dict):
-            all_objects.extend(s["sensors"])
+            if "sensors" in s:
+                all_objects.extend(s["sensors"])
+            if "sensor" in s:
+                all_objects.append(s["sensor"])
```

Comment:
This is in principle a generic util function. What about cases other than `sensors_to_show`? 

In this implementation, we silently drop all dict content other than the ones in "sensor" and "sensors" keys.
How would we flatten a dict?

I know this is a bugfix PR, but we should make the docstring a little better here.

And I did a little research - it seems we are not using this function for anything else than `sensors_to_show` (once for something that we know is already a flat list). So (in a follow-up PR), we could actually move this function next to the SensorsToShowSchema, or in models/GenericAsset, where it makes more sense.

Reply:
Yes, i found that `flatten_unique` is only used for `sensors_to_show`  that's why i added that specification to `sensor` and `sensors`. i agree moving to models/GenericAsset or next to SensorsToShowSchema would be better. 

Do you mean moving this function as-is next to `SensorsToShowSchema` or in `models/GenericAsset`, or are you suggesting making it a class attribute (method) of one of these classes?

Reply:
Well, I found [one occurence](https://github.com/FlexMeasures/flexmeasures/blob/main/flexmeasures/data/models/generic_assets.py#L754) where flatten_unique is not used explicitly for `sensors_to_show`, but it might not be needed there, as we expect already a flat list.

I think we can move it as-is. Maybe renaming it to `flatten_sensors_to_show`? Hm, maybe `SensorsToShowSchema._flatten()` would work (be sure to add a docstring)